Add min_p sampling

pau-mensa opened the pull request:
## Summary:
This pull request introduces min_p sampling in both main.py and torch_main.py, and top_k sampling in torch_main.py.

## Changes:
### min_p Sampling:

- Added min_p sampling to the sample function in main.py and torch_main.py.

- The min_p parameter is now an optional argument in the sample function, with a default value of 0.0 (no min_p constraint).

### top_k Sampling:

- Added top_k sampling to the sample function in torch_main.py, since it seemed to be missing.

xjdr-alt commented:
Sorry about the timing, but i added the entropy sampler which breaks your commit. Would you mind rebasing and retesting?

pau-mensa commented:
No problem! I rebased and everything seems to check out. I've added the sampling params of the sample function to the _sample call as well.

Since #6 mentioned a variable name that could lead to misinterpretations (top_p) and suggested changing it to p_max I looked up the naming in the transformers library for the min_p implementation, which is top_probs. I've decided to keep the same name, but feel free to change it if you want to.

xjdr-alt commented:
Awesome, thank you! Incorporated into https://github.com/xjdr-alt/entropix/pull/11
